### PR TITLE
Compilation adjustments for libcsprrpc

### PR DESCRIPTION
**Pull request by TomaszAugustyn**
Compilation adjustments for libcsprrpc:
1. Fixes in CMakeLists.txt
- `cryptopp-static` -> `${CRYPTOPP_LIBRARIES}`
- `find_package(cryptopp CONFIG REQUIRED)` -> `find_package(CryptoPP REQUIRED)`
- order of commands

2. Adjustements for bitwise operations on enums

### CODE REVIEW

**Review by bartalke — APPROVED** [review]

**Review by rrozek — APPROVED** [review]

File: lib/cryptopp-pem/CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,6 @@ set(SOURCES
 add_library(cryptopp_pem STATIC ${SOURCES})
 target_include_directories(cryptopp_pem PUBLIC ${SRC_PATH})
 target_link_libraries(cryptopp_pem
-    PUBLIC 
-    cryptopp-static
+    PUBLIC
+    ${CRYPTOPP_LIBRARIES}
```
Comment (polishcode) [inline lib/cryptopp-pem/CMakeLists.txt:16]: Unfortunately ${CRYPTOPP_LIBRARIES} is not set in windows, thus linking fails. We need to discuss it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -15,11 +15,26 @@ else()
...
-#TODO the rest of deps should be moved outside of the project too (if possible)
+# TODO the rest of deps should be moved outside of the project too (if possible)
 MESSAGE("using deps from vcpkg or system")
 find_package(Boost 1.75.0 REQUIRED)
 find_package(OpenSSL REQUIRED)
-find_package(cryptopp CONFIG REQUIRED)
+find_package(CryptoPP REQUIRED)
+
+# TODO remove this logging
+message("CRYPTOPP_LIBRARIES: " ${CRYPTOPP_LIBRARIES})
```
Comment (bartalke) [inline CMakeLists.txt:25]: I guess it is remove ? :)

File: CMakeLists.txt [hunk 3]
```diff
@@ -15,11 +15,26 @@ else()
...
+# in such a case just find it manually
+if(NOT CRYPTOPP_LIBRARIES)
+	# TODO remove this logging
+	message("looking for CRYPTOPP_LIBRARIES...")
+	find_library(CryptoPP_LIBRARY NAMES cryptopp-static cryptopp DOC "CryptoPP library")
+	set(CRYPTOPP_LIBRARIES ${CryptoPP_LIBRARY})
+endif()
+
+# TODO remove this logging
+message("CRYPTOPP_LIBRARIES: " ${CRYPTOPP_LIBRARIES})
```
Comment (bartalke) [inline CMakeLists.txt:37]: To rm ? :)

**Review by bartalke — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (polishcode) [reply CMakeLists.txt:25]: @TomaszAugustyn will check if everything works correct tomorrow, then please remove.

File: CMakeLists.txt (hunk 3 above)
Comment (polishcode) [reply CMakeLists.txt:37]: @TomaszAugustyn will check if everything works correct tomorrow, then please remove.

File: lib/cryptopp-pem/CMakeLists.txt (hunk 1 above)
Comment (TomaszAugustyn) [reply lib/cryptopp-pem/CMakeLists.txt:16]: Done.

File: CMakeLists.txt (hunk 2 above)
Comment (TomaszAugustyn) [reply CMakeLists.txt]: Removed.

File: CMakeLists.txt (hunk 3 above)
Comment (TomaszAugustyn) [reply CMakeLists.txt]: Removed.

**Pull request merged**